### ISSUE TITLE
Cudnn batch norm kernel (batchnorm_bwtr_nhwc_semiPersist) gets blocked by overlapping NCCL all_reduce calls

### ISSUE THREAD

**Issue by zhengwy888** [pytorch/pytorch#80259]
### 🐛 Describe the bug

When cudnn + channels_last format is enabled, under distributed DDP, the backward pass of cudnn batchnorm backward can be stalled by the overlapped nccl all_reduce call. 

Sorry I don't have a code snippet for repro, but here are some screenshots of torch profiler

Blocking batch_norm_backward
![image](https://user-images.githubusercontent.com/1190730/175750442-13d30ee6-5c8b-4f59-bbcd-95027448af5b.png)

Neither the aten native implementation nor the NCHW version of cudnn kernel `bn_bw_1C11_singleread_specialized` shows this behavior. 

Technically this seems like a cuDNN bug, but I don't know where/how to file a cudnn bug, so filing it here instead. This issue is also reported on https://github.com/NVIDIA/nccl/issues/338 https://github.com/NVIDIA/nccl/issues/661

### Versions

```
Collecting environment information...
PyTorch version: 1.11.0a0+gitbc2c6ed
Is debug build: False
CUDA used to build PyTorch: 11.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: 14.0.5 (https://github.com/conda-forge/clangdev-feedstock 0f793d7b2ad6a3c631eb76429f4e4def37c890dd)
CMake version: version 3.19.1
Libc version: glibc-2.31

Python version: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:10)  [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-64-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.4.120
GPU models and configuration: GPU 0: NVIDIA A100-SXM-80GB
Nvidia driver version: 470.57.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.19.5
[pip3] pytorch3d==0.6.1
[pip3] torch==1.11.0+bc2c6ed.cuda114.cudnn841.se02.ap
[pip3] torch-scatter==2.0.8
[pip3] torch-tb-profiler==0.4.0
[pip3] torchfile==0.1.0
[pip3] torchvision==0.9.0a0+8fb5838
[conda] magma-cuda111             2.5.2                         1    pytorch
[conda] mkl                       2020.4             h726a3e6_304    conda-forge
[conda] mkl-include               2022.1.0           h84fe81f_915    conda-forge
[conda] numpy                     1.19.5           py38h8246c76_3    conda-forge
[conda] pytorch3d                 0.6.1                    pypi_0    pypi
[conda] torch                     1.11.0+bc2c6ed.cuda114.cudnn841.se02.ap          pypi_0    pypi
[conda] torch-scatter             2.0.8                    pypi_0    pypi
[conda] torch-tb-profiler         0.4.0                    pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchvision               0.9.0a0+8fb5838          pypi_0    pypi
```

**Comment by ptrblck** [comment]
This is unfortunately expected behavior as NCCL will spin a CUDA kernel for its communications (and thus use >=1 sm) which could prevent other kernel launches which require high compute resources (large sm count) such as the persistent NHWC cuDNN batchnorm layer.
cuDNN does not have an API to limit the sm count, which might be a way to avoid these waiting times.

**Comment by zhengwy888** [comment]
One of the workaround is to use native_batch_norm_backward instead of cudnn_batch_norm_backward, from my test it seems this combination will preserve most of the throughput improvement of cudnn + channels_last without introducing a blocking kernel. Do you think this 'hack' is worth implementing at the pytorch level? Or we should wait until cudnn gives some control on SM count like you mentioned.

**Comment by ptrblck** [comment]
That's indeed an interesting observation and it would be even more interesting to see how code-generated normalization kernels would perform in this use case. 

CC @csarofeen @kevinstephano in case we want to profile the nvfuser implementation for it.

As for the missing cuDNN API: we can surely follow up on this feature request (we were already talking about it), but I would not expect to see an implementation of this API soon.

**Comment by aazzolini** [comment]
Hey all, is this still a limitation of CuDNN API at the moment? I am hitting this issue right now.
